sema/commons/j2/j2_functions.py

```
@@ -46,43 +52,77 @@ def xsd_format_boolean(content, quote, suffix):
-def xsd_format_date(content, quote, suffix):
+def xsd_format_date(content: any, quote: str, *_) -> str:
     # make rigid date
+    if isinstance(content, datetime):
+        raise ValueError(
+            "use datetime format for datetime values, or past .date() result"
+        )
+
     if not isinstance(content, date):
         asdt = parser.isoparse(content).date()
     else:
         asdt = content
```

Ruff (0.8.2)

Use ternary operator `asdt = parser.isoparse(content).date() if not isinstance(content, date) else content` instead of `if`-`else`-block

Replace `if`-`else`-block with `asdt = parser.isoparse(content).date() if not isinstance(content, date) else content`

(SIM108)